site/docs/pages/token/token-search.mdx

```diff
@@ -0,0 +1,62 @@
+import { TokenSearch } from '../../../../src/token/components/TokenSearch';
+import App from '../App';
+import { tokens } from '../../components/tokens';
+
+# `<TokenSearch />`
+
+### Warning: component under development, may change in coming weeks
+
+The `<TokenSearch />` component allows users to search on a given list of tokens by name, symbol or address with an optional debounce delay.
+
+## Usage
+
+:::code-group
+
+```tsx [code]
+import { TokenSearch } from '@coinbase/onchainkit/token';
+
+const tokens = [
+  // ...
+];
+
+// example of async onChange handler
+const handleChange = useCallback((value) => {
+  async function getData(value) {
+    const tokens = await getTokens({ search: value });
+
+    // Do something with tokens here
+  }
+
+  getData(value)
+}, [])
+
+// example of onChange handler with a fixed list
+const handleChange = useCallback((value) => {
+  const filteredTokens = tokens.filter(({ name, symbol, address }) => {
+    const v = value.toLowerCase();
+    return (
+      name.toLowerCase().includes(v) ||
+      symbol.toLowerCase().includes(v) ||
+      address.toLowerCase().includes(v)
+    );
+  });
+
+  // Do something with filteredTokens here
+}, [tokens]);
+
+<TokenSearch onChange={handleChange} delayMs={200} />; // [!code focus]
```

Comment:
@Zizzamia how about this?

Reply:
Oh very interesting. I think we can show the example in two code examples.